core/src/main/scala/com/raphtory/internals/management/Scheduler.scala

```diff
@@ -0,0 +1,44 @@
+package com.raphtory.internals.management
+
+import cats.effect.IO
+import cats.effect.unsafe.IORuntime
+import com.raphtory.internals.components.Component
+import com.typesafe.scalalogging.Logger
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+/** @note DoNotDocument */
+private[raphtory] class Scheduler {
+  //FIXME: wipe this class out as we move to cats-effect
+  private val threads: Int   = 8
+  private val logger: Logger = Logger(LoggerFactory.getLogger(this.getClass))
+
+  implicit val runtime = IORuntime.global
+
+  def execute[T](component: Component[T]): Unit =
+    IO.delay(component.run()).unsafeToFuture()
+
+  def scheduleOnce(delay: FiniteDuration, task: => Unit): () => Future[Unit] = {
+    val (_, cancel) = (IO.sleep(delay) *> IO(task)).unsafeToFutureCancelable()
+    cancel
+  }
+
+  def executeInParallel(
+      tasks: Iterable[IO[Unit]],
+      onSuccess: => Unit,
+      errorHandler: (Throwable) => Unit
+  ): () => Future[Unit] = {
+
+    val (_, cancelable) = IO
+      .parSequenceN(threads)(tasks.to(LazyList))
+      .onError(t => IO(errorHandler(t)))
+      .flatMap(_ => IO(onSuccess))
+      .unsafeToFutureCancelable()
+
+    cancelable
+  }
+
+  def shutdown(): Unit = {}
```

Comment:
Is this just here as a stub at the moment to be filled in? We had the old one to attempt to kill off everything that the scheduler had started, but I imagine this may not be necessary with the planned changes. Could possibly remove as is only called in the GraphDeployment and TemporalGraphConnection - with the latter solely existing to do this function call so the class could probably go as well.

Reply:
yes, I'll take it out